src/ProxySettingsMac.mm

```diff
@@ -75,14 +116,18 @@ PROXY_PROTO parseProxyType(CFStringRef type)
    return PP_UNSUPPORTED;
 }
 
-bool isDirect(CFDictionaryRef proxy)
+PROXY_PROTO readProxyType(CFDictionaryRef dict)
 {
-    return PP_NONE == parseProxyType(reinterpret_cast<CFStringRef>(CFDictionaryGetValue(proxy, kCFProxyTypeKey)));
+    CFDictionaryRef cf = NULL;
+    if (!CFDictionaryGetValueIfPresent(dict, kCFProxyTypeKey, reinterpret_cast<const void**>(&cf))) {
+        return PP_UNSUPPORTED;
+    }
+    return parseProxyType(reinterpret_cast<CFStringRef>(cf));
 }
 
-bool isProxySupported(CFDictionaryRef proxy)
+bool isProxySupported(PROXY_PROTO proxyType)
 {
-    return PP_HTTP == parseProxyType(reinterpret_cast<CFStringRef>(CFDictionaryGetValue(proxy, kCFProxyTypeKey)));
+    return PP_HTTP == proxyType || PP_HTTPS == proxyType || PP_PAC == proxyType;
```

Comment:
i'd prefer not to have yoda style comparisons, but we don't have a firm standard